game/scripts/vscripts/items/aghanims.lua

```diff
@@ -135,34 +126,24 @@ function modifier_item_aghanims_talents:SetTalents(tree)
       'Trying to update talent but talent choice was let through!'
     )
 
-    -- print('At leve ' .. level .. ' hero chose ' .. parent['talentChoice' .. level])
-
-    if self:IsStatsTalent(leftAbility:GetName()) and parent['talentChoice' .. level] == 'right' then
-      talentOverrides[#talentOverrides + 1] = leftAbility:GetName()
-    elseif self:IsStatsTalent(rightAbility:GetName()) and parent['talentChoice' .. level] == 'left' then
-      talentOverrides[#talentOverrides + 1] = rightAbility:GetName()
+    if claim then
+      if leftLevel == 0 then
+        leftAbility:SetLevel(1)
```

Comment:
I tested and the way he implemented worked on Undying CD and also worked on CK stats (all stats and str ) and also others HP regen and Mana regen talents.
The only difference is that you lose the stats if you drop the aghs

Reply:
I tested and the way he implemented worked on Undying CD and also worked on CK stats (all stats and str ) and also others HP regen and Mana regen talents.
The only difference is that you lose the stats if you drop the aghs

Reply:
Yeah, I was going to change it to `UpgradeAbility` cause I assumed `SetLevel` wouldn't trigger the learn events or something, but I've tested this and it seems to work fine. Even when dropping and picking up Agh's. I'm not really sure why QoP's CD is even bugged actually. Dunno if it's only when you drop Agh's so the level goes back to 0. That or it acts differently on dedicated servers for some reason.

The issue with Undying was half that the modifier got removed when he died. Making the Talent level 0 whenever he died and causing it to not start cooldown. Just fixing that would allow you to just drop Agh's to have permanent level 0, though. You'd lose some other Talents, sure, but you'd be immortal.

Reply:
Interesting, I wonder why stats work now but didn't before. Pretty weird. Must have to do with the modifier being added and removed causing the talents to be left in broken states...